evm/utils/state.py

```diff
@@ -30,3 +37,16 @@ def diff_state_db(expected_state, state_db):
                     actual_storage_value,
                     expected_storage_value,
                 )
+
+
+# Make the root of a receipt tree
+def make_trie_root_and_nodes(transactions):
```

Comment:
Since it looks like we're introducing binary tries, do you think it's worth changing the name of this DB to something like `make_hexary_trie_root_and_nodes`?

Reply:
How about passing Trie class as a parameter? Otherwise, we will need to refactor and implement another `make_binary_trie_root_and_nodes` later.
 
```python
def make_trie_root_and_nodes(transactions, trie_class=HexaryTrie):
    chaindb = BaseChainDB(MemoryDB())
    db = chaindb.db
    transaction_db = trie_class(db)
    ...
```